docs/reference/enable-gpu.rst

```diff
@@ -1,46 +1,153 @@
-Enable Nvidia GPUs
+.. _nvidia-gpu:
+
+Enable NVIDIA GPUs
 ==================
 
-To be able to run workloads on GPU with DSS, you need to have:
+The following section describes how to configure the DSS to utilize the host's
+NVIDIA GPU. This is achieved by configuring the underlying MicroK8s, that the
+DSS is relying for running the containerised workloads.
+
+Prerequisites
+^^^^^^^^^^^^^
+
+* Have run the :ref:`tutorial` tutorial
+
+Install the NVIDIA Operator
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To ensure the DSS can utilise the NVIDIA GPUs:
+
+1. The host will need to have the NVIDIA drivers installed
+2. MicroK8s will need to be setup to utilise those drivers
+
+MicroK8s is leveraging the `NVIDIA Operator`_ to for setting up and
+configuring the NVIDIA runtime. The NVIDIA Operator will also install
+the NVIDIA drivers, if they are not present already on the host machine.
+
+To enable the NVIDIA runtimes on MicroK8s, run the following command:
+
+.. code-block:: bash
+
+   sudo microk8s enable gpu
+
+.. note::
+   The NVIDIA Operator will detect if the NVIDIA drivers are not present at all
+   and will install them in this case. But the NVIDIA Operator can also detect
+   if you have drivers already installed. In this case the NVIDIA Operator will
+   use the host's drivers.
+
+Verify the NVIDIA Operator is up
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before spinning up workloads that you need to ensure the GPU Operator has
+been successfully initialized. This can take around 5 minutes.
+
+Ensure DaemonSet is Ready
+"""""""""""""""""""""""""
+
+First we need to ensure that the DaemonSet for the Operator Validator is
+created:
+
+
+.. code-block:: bash
+
+  while ! sudo microk8s.kubectl get ds -n gpu-operator-resources nvidia-operator-validator
+  do
+    sleep 5
+  done
+
+.. note::
+   It will take some seconds for the DaemonSet to get created. The above command will
+   return a message like the following at the beginning:
+   ``Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found``
+
+The expected output should be:
+
+.. code-block:: text
+
+   Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found
+   ...
+   Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found
+   NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                                   AGE
+   nvidia-operator-validator   1         1         0       1            0           nvidia.com/gpu.deploy.operator-validator=true   17s
+
+Ensure the Validator succeeded
+""""""""""""""""""""""""""""""
+
+Next we'll need to wait for the Validator Pod to succeed:
```

Comment:
```suggestion
Next, you have to wait until the Validator Pod to succeed:
```
If this and previous steps are a sequence, I would create the titles with numbering, i.e., 
1. Ensure ...
2. Wait for the...
Not sure yet if headings usually include numbered lists (need to check other product docs), but I would go ahead with this solution.

Reply:
Hmm but it's already as a list. Keep in mind that the `Ensure the DaemonSet succeeded` is only one wait step. That's why the `Ensure the DaemonSet succeeded` is one step.

And similarly `Ensure the Validator succeeded` is another step, and inside it's just a waiting command

![image](https://github.com/canonical/data-science-stack/assets/11134742/1d804ed3-2664-4830-bec0-e82d25968190)

Reply:
What I meant here is to something similar to these [numbered lists](https://canonical-data-science-stack.readthedocs-hosted.com/en/latest/how-to/jupyter-notebook/create-notebook/#creating-a-notebook). In the sense that, to verify the NVIDIA Operator is up, you have to do:
1. Ensure DaemonSet is ready
2. Ensure the Validator succeeded

Reply:
I'm not really a fan of the identation in this page, and would prefer to use headers in case there needs to be a lot of text and/or code blocks in a numbered list.

Would you be OK with adding numbers at the beginning of the headers, like `1. Ensure DaemonSet is ready` and `2. Ensure the Validator succeeded` in the headings?

Or you'd also hugely prefer to have the content indented for the numbered list?

Reply:
Alright, let's keep it as is. I am not happy with numbering headings, I had a quick look and we don't do it in other Canonical docs.